packages/analytics-js/src/components/userSessionManager/utils.ts

```diff
@@ -1,8 +1,94 @@
-import { ApiObject } from '@rudderstack/analytics-js/state/types';
+import { state } from '@rudderstack/analytics-js/state';
+import { ApiObject, SessionInfo } from '@rudderstack/analytics-js/state/types';
 import { Nullable } from '@rudderstack/analytics-js/types';
+import { ILogger } from '@rudderstack/analytics-js/services/Logger/types';
+import { DEFAULT_SESSION_TIMEOUT } from '@rudderstack/analytics-js/constants/timeouts';
+import { countDigits } from '../utilities/number';
 
-const isValidTraitsValue = (traits?: Nullable<ApiObject>) => {
-  return traits !== undefined && traits !== null && Object.keys(traits).length > 0;
+const MIN_SESSION_ID_LENGTH = 10;
+
+/**
+ * A function to check the input object is not empty, undefined or null
+ * @param obj input object
+ * @returns boolean
+ */
+const hasValidValue = (obj?: Nullable<ApiObject>) =>
+  obj !== undefined && obj !== null && Object.keys(obj).length > 0;
+
+/**
+ * A function to validate current session and return true/false depending on that
+ * @param {number} timestamp
+ * @returns boolean
+ */
+const isValidSession = (timestamp: number): boolean =>
+  Boolean(
+    state.session.sessionInfo.value.expiresAt &&
```

Comment:
utility methods should not access state directly, should be small reusable code if possible, pass the state value as argument expiresAt and have timestamp as seconf optional argument with default have to now

Reply:
Addressed